### PR TITLE
Update alteration.py with AI and CCPI

### PR DESCRIPTION
**Pull request by NicolasPietteLauziere**
## Description
Added Alteration index (Ishikawa et al., 1976) and Chlorite-carbonate-pyrite index (Large, Gemmell, and Paulick, 2001).

## Related Issue
No issue

## Motivation and Context
These alteration index are used for mineral exploration of VMS deposits using whole rock geochemistry.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by morganjwilliams** [comment]
Hi @NicolasPietteLauziere, thanks for opening a pull request! I'll check it out and get back to you shortly.

One first thing - could you update the pull request to use `develop` as a base branch, I only use `master` for new releases and the odd hotfix.

File: pyrolite/geochem/alteration.py [hunk 1]
```diff
@@ -127,3 +127,35 @@ def WIP(df: pd.DataFrame):
...
+
+    References
+    ----------
+    .. [#ref_1] Ishikawa Y, Sawaguchi T, Iwaya S, Horiuchi M (1976). Delineation of prospecting targets for Kuroko deposits based on modes of volcanism of underlying dacite and alteration halos.
+           Mining Geology 27:106-117.
+           
+    """
+    return 100*(df.K2O + df.MgO) / (df.K2O + df.MgO + df.Na2O + df.CaO)
+
+def CCPI(df: pd.DataFrame):
```
Comment (morganjwilliams) [inline pyrolite/geochem/alteration.py:144]: Might want to add the `@update_docstring_references` decorator here too.

File: pyrolite/geochem/alteration.py [hunk 2]
```diff
@@ -127,3 +127,35 @@ def WIP(df: pd.DataFrame):
...
+    """
+    Chlorite-carbonate-pyrite index of Large et al. (wt%) [#ref_1]_
+    FeO is total FeO+Fe2O3
+
+    References
+    ----------
+    .. [#ref_1] Large R R, Gemmell, J B, Paulick, H(2001). The alteration box plot: A simple approach to understanding the relationship between alteration mineralogy and lithogeochemistry associated with volcanic-hosted massive sulfide deposits.
```
Comment (morganjwilliams) [inline pyrolite/geochem/alteration.py:151]: Condense the references to fit within about 80-88 characters per line if possible; this makes it easy to read on smaller screens.

File: pyrolite/geochem/alteration.py [hunk 3]
```diff
@@ -127,3 +127,35 @@ def WIP(df: pd.DataFrame):
 
     """
     return 2 * df.Na2O / 0.35 + df.MgO / 0.9 + 2 * df.K2O / 0.25 + df.CaO / 0.7
+
+@update_docstring_references
+def AI(df: pd.DataFrame):
```
Comment (morganjwilliams) [inline pyrolite/geochem/alteration.py:132]: Might want to rename this one; having a function called `AI` might be a bit loaded! Perhaps `IAI` or even `IshikawaAltIndex` if you'd like to be explicit.

File: pyrolite/geochem/alteration.py [hunk 4]
```diff
@@ -127,3 +127,35 @@ def WIP(df: pd.DataFrame):
...
+def CCPI(df: pd.DataFrame):
+    """
+    Chlorite-carbonate-pyrite index of Large et al. (wt%) [#ref_1]_
+    FeO is total FeO+Fe2O3
+
+    References
+    ----------
+    .. [#ref_1] Large R R, Gemmell, J B, Paulick, H(2001). The alteration box plot: A simple approach to understanding the relationship between alteration mineralogy and lithogeochemistry associated with volcanic-hosted massive sulfide deposits.
+           Economic Geology 96:957-971.
+           doi:`10.2113/gsecongeo.96.5.957`
```
Comment (morganjwilliams) [inline pyrolite/geochem/alteration.py:153]: Can add the hyperlink here if you'd like: ```doi:`10.2113/gsecongeo.96.5.957 <http://dx.doi.org/10.2113/gsecongeo.96.5.957>`__```.

File: pyrolite/geochem/alteration.py [hunk 5]
```diff
@@ -127,3 +127,35 @@ def WIP(df: pd.DataFrame):
...
+           
+    """
+    if 'FeO' in df.columns:
+        ccpi = 100*(df.MgO + df.FeO) / (df.MgO + df.FeO + df.Na2O + df.K2O)
+    elif 'Fe2O3'in df.columns and 'FeO' in df.columns:
+        ccpi = 100*(df.MgO+ df.FeO + df.Fe2O3) / (df.MgO+ df.FeO + df.Fe2O3 + 
+                                                  df.Na2O + df.K2O)
+    return(ccpi)
```
Comment (morganjwilliams) [inline pyrolite/geochem/alteration.py:161]: Don't need the enclosing brackets here; but it'll all be formatted away with `black` after merge, so no particular issue.

File: test/geochem/geochem_alteration.py [hunk 6]
```diff
@@ -87,5 +87,39 @@ def test_WIP(self):
         df.loc[:, "WIP"] = WIP(df)
 
 
+if __name__ == "__main__":
```
Comment (morganjwilliams) [inline test/geochem/geochem_alteration.py:90]: We only need the `unittest.main()` call once per file, and there should already be one at the end, so feel free to remove the extras here.

**Comment by NicolasPietteLauziere** [comment]
Thanks for the comments Morgan! I did the edits and pushed it to the pull request.

**Comment by morganjwilliams** [comment]
@NicolasPietteLauziere did you mean to close this one? If not let me know and we can probably merge these in shortly.

**Comment by NicolasPietteLauziere** [comment]
Yes they are. Sorry about that, I thought that "closing" a request meant that the updates were ready to be merged!

**Comment by morganjwilliams** [comment]
Thanks @NicolasPietteLauziere! Looks like a few of the tests might be failing due to one or two of lines related to dataframe assignment; I'll tag them shortly.

File: test/geochem/geochem_alteration.py [hunk 7]
```diff
@@ -87,5 +87,33 @@ def test_WIP(self):
...
+
+    def setUp(self):
+        self.cols = ["SiO2", "CaO", "MgO", "FeO", "TiO2", "Na2O", "K2O", "Al2O3"]
+        self.df = pd.DataFrame(
+            {k: v for k, v in zip(self.cols, np.random.rand(len(self.cols), 10))}
+        )
+
+    def test_CCPI(self):
+        df = self.df
+        df.loc[:, "CCPI"] = CCPI(df)
```
Comment (morganjwilliams) [inline test/geochem/geochem_alteration.py:115]: A test will fail here due to some changes in recent `pandas` versions. You can update it with:
```python
df["CCPI"] = CCPI(df)
```

File: test/geochem/geochem_alteration.py [hunk 8]
```diff
@@ -87,5 +87,33 @@ def test_WIP(self):
...
+
+    def setUp(self):
+        self.cols = ["SiO2", "CaO", "MgO", "FeO", "TiO2", "Na2O", "K2O", "Al2O3"]
+        self.df = pd.DataFrame(
+            {k: v for k, v in zip(self.cols, np.random.rand(len(self.cols), 10))}
+        )
+
+    def test_AI(self):
+        df = self.df
+        df.loc[:, "AI"] = AI(df)
```
Comment (morganjwilliams) [inline test/geochem/geochem_alteration.py:101]: A test will fail here due to some changes in recent `pandas` versions. You can update it with:
```python
df["AI"] = AI(df)
```

**Comment by morganjwilliams** [comment]
@NicolasPietteLauziere I've merged this into `develop` and will update a few of the tests across the board for `pyrolite.geochem.alteration` (with corresponding API docs [here](https://pyrolite.readthedocs.io/en/develop/api/geochem/alteration.html#pyrolite.geochem.alteration.IshikawaAltIndex)). I'll also add you to the changelog and contributors list - let me know if you're happy for me to link to your GitHub/Twitter/other profile there. pyrolite v0.3.1 is just around the corner, so expect this to be released soon; let me know if you spot anything which looks off in the meantime!

**Comment by NicolasPietteLauziere** [comment]
@morganjwilliams  thanks for adding me to the contributors.

**Pull request merged**